### PR TITLE
[SPARK-45485][CONNECT] User agent improvements: Use SPARK_CONNECT_USER_AGENT env variable and include environment specific attributes

### PR DESCRIPTION
**Pull request by dillitz**
### What changes were proposed in this pull request?
With this PR similar to the[ Python client](https://github.com/apache/spark/blob/2cc1ee4d3a05a641d7a245f015ef824d8f7bae8b/python/pyspark/sql/connect/client/core.py#L284) the Scala client's user agent now:

1. Uses the SPARK_CONNECT_USER_AGENT environment variable if set
2. Includes the OS, JVM version, Scala version, and Spark version

### Why are the changes needed?
Feature parity with the Python client. Better observability of Scala Spark Connect clients.

### Does this PR introduce _any_ user-facing change?
By default, the user agent string now contains more useful information.
Before:
`_SPARK_CONNECT_SCALA`
After:
`_SPARK_CONNECT_SCALA spark/4.0.0-SNAPSHOT scala/2.13.12 jvm/17.0.8.1 os/darwin`

### How was this patch tested?
Tests added & adjusted.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
My bad, there was a PR merged with a wrong PR ticket. I switched the ticket each other.

File: connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala [hunk 1]
```diff
@@ -585,6 +587,27 @@ object SparkConnectClient {
...
+      else if (os.contains("linux")) "linux"
+      else if (os.contains("win")) "windows"
+      else "unknown"
+    }
+    List(
+      value,
+      s"spark/$SPARK_VERSION",
+      s"scala/$scalaVersion",
+      s"jvm/$jvmVersion",
+      s"os/$osName").mkString(" ")
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:608]: Might not be a big deal but would be great to keep them same if it's easy to make it consistent

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:608]: Ah, okie. you're doing it above 598L

**Review by HyukjinKwon — APPROVED** [review]
Seems fine given that it's consistent

File: connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (dillitz) [reply connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala:608]: OS should be consistent for Mac, Windows, and Linux.

**Comment by dillitz** [comment]
FYI @grundprinzip

**Comment by HyukjinKwon** [comment]
Lemme merge this as it seems pretty consistent with the exiting codebase.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**